waiter/src/waiter/scaling.clj

```diff
@@ -429,8 +429,8 @@
                         state-chan
                         (let [{:keys [service-id->healthy-instances service-id->unhealthy-instances service-id->expired-instances]} args
                               existing-service-ids-set (set (keys service-id->router-state))
-                              service-ids-set (set/union (-> service-id->healthy-instances (keys) (set))
-                                                         (-> service-id->unhealthy-instances (keys) (set)))
+                              service-ids-set (set (concat (keys service-id->healthy-instances)
+                                                           (keys service-id->unhealthy-instances)))
```

Comment:
How about `(into (-> service-id->healthy-instances keys set) (-> service-id->unhealthy-instances keys)` to help us avoid the intermediate lazy sequence.

Reply:
Using `concat` was faster than creating the intermediate sets and then doing a union, but I ran another microbenchmark and found that using `into` as you suggested results in performance as good as if not better than the `concat` version. That surprised me, since I assumed the `concat` version would perform better since `concat` itself is very low overhead. Apparently accessing the elements through the lazy sequence layer adds a non-trivial amount of overhead.